lib/transmit.js

```diff
@@ -285,6 +285,11 @@ internals.transmit = function (response, callback) {
         }
     }
 
+    const isInjection = Shot.isInjection(request.raw.req);
+    if (!isInjection && !request.connection._started) {
+        request.raw.res.setHeader('connection', 'close');
```

Comment:
Since injections apply to a non-started server, it would add this header to every request. Further, injections are not connection oriented and it doesn't make sense to mess with this header.